lib/widgets/search-friends.dart

```diff
@@ -1,15 +1,150 @@
+import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:firebase_database/firebase_database.dart';
+import 'package:firebase_auth/firebase_auth.dart';
 
-class SearchFriends extends StatelessWidget {
+class SearchFriends extends StatefulWidget {
+  @override
+  _SearchFriendsState createState() => _SearchFriendsState();
+}
+
+class _SearchFriendsState extends State<SearchFriends> {
   final databaseReference = FirebaseDatabase.instance.reference();
+  Map result;
+  bool noFound = false;
 
   @override
   Widget build(BuildContext ctx) {
-    databaseReference.once().then((DataSnapshot snapshot) {
-      print('Data : ${snapshot.value}');
-    });
+    return new Padding(
+      padding: const EdgeInsets.all(16.0),
+      child: new Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Text("Add friend",
+              style:
+                  new TextStyle(fontSize: 24.0, fontWeight: FontWeight.bold)),
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: 16, horizontal: 8),
+            child: new TextField(
+                onSubmitted: searchFriend,
+                autocorrect: false,
+                textCapitalization: TextCapitalization.none,
```

Comment:
Is there a way to clear previous results onChange of input? Currently, if I searched for xyz@xyz.com found it, then type abc@abc.com, xyz@xyz still shows in results prior to pressing enter.